main/src/ca/uwaterloo/flix/language/phase/Safety.scala

```diff
@@ -514,53 +501,66 @@ object Safety {
         acc + (signature -> m)
     }
   }
+  
+  /**
+    * Convert a `java.lang.reflect.Method` to a MethodSignature.
+    */
+  private def getJavaMethodSignature(method: java.lang.reflect.Method) = {
+    MethodSignature(method.getName(), 
+      getFlixType(method.getReturnType),
+      method.getParameterTypes().toList.map(getFlixType))
+  }
+
+  /**
+    * Given a class, get sets representing both the methods that can be implemented,
+    * and the methods that must be implemented.
+    */
+  private def getJavaMethods(clazz: java.lang.Class[_]): (Set[MethodSignature], Set[MethodSignature]) = {
```

Comment:
Given a class or interface, returns a pair of the methods that *can* be implemented and that *must* be implemented. 

- A (non-static) method *can* be implemented if it exists in the class.
- A (non-static) method *must* be implemented if it is abstract.

Reply:
If you update this I can do a merge now 👍